presto-hive/src/main/java/com/facebook/presto/hive/ParquetRecordWriterUtil.java

```diff
@@ -39,12 +41,14 @@ public final class ParquetRecordWriterUtil
     private static final Field REAL_WRITER_FIELD;
     private static final Field INTERNAL_WRITER_FIELD;
     private static final Field FILE_WRITER_FIELD;
+    private static final AtomicReference<Field> COLUMN_WRITE_STORE_FIELD;
 
     static {
         try {
             REAL_WRITER_FIELD = ParquetRecordWriterWrapper.class.getDeclaredField("realWriter");
             INTERNAL_WRITER_FIELD = ParquetRecordWriter.class.getDeclaredField("internalWriter");
             FILE_WRITER_FIELD = INTERNAL_WRITER_FIELD.getType().getDeclaredField("parquetFileWriter");
+            COLUMN_WRITE_STORE_FIELD = new AtomicReference<>(null);
```

Comment:
The `columnStore` field exists in the `InternalParquetRecordWriter` class which is declared as package-private. This means that we don't have the means to even access the class directly in the codebase. However, we can get the `Class<T>` instance from an existing reference. However, that can't be done until we initialize at least one writer.

Use an atomic reference and do a CAS when creating Parquet writers to initialize the `Field` instance if it hasn't been already

Reply:
I think this is fine, but it definitely deserves a comment that basically says this so people aren't scratching their heads years down the road why the code is written in this way.

Reply:
added